docs/how-to/extending/unbound_expression.qmd

```diff
@@ -0,0 +1,101 @@
+# Write and execute unbound expressions
+
+One of the most powerful features of Ibis is the separation of transformation
+logic from the execution engine, which allows you to "write once, execute
+everywhere".
+
+## Unbound tables
+
+In Ibis, you can define unbound tables. An unbound table is a table with a
+specified schema but not connected to a data source. You can think of it as an
+empty spreadsheet with just the header. Even though the spreadsheet is empty,
+you know what the data would look like.
+
+Unbound tables allow you to write transformations for data as long as it
+conforms to the provided schema. You don't need to connect to data until you're
+ready to execute the expression and compute outputs.
+
+## Example
+
+Here's how we can define an unbound table in Ibis:
+
+```{python}
+import ibis
+
+schema = {
+    "carat": "float64",
+    "cut": "string",
+    "color": "string",
+    "clarity": "string",
+    "depth": "float64",
+    "table": "float64",
+    "price": "int64",
+    "x": "float64",
+    "y": "float64",
+    "z": "float64",
+}
+diamonds = ibis.table(schema, name="diamonds")
+diamonds
+```
+
+So far, we have an empty `diamonds` table that contains 10 columns. Even though
+there is no data in the `diamonds` table right now, we can write
+transformations knowing that these are the columns available to us.
+
+Given this table of diamonds of various carats, cuts, and colors, we're
+interested in learning the average carat for each color of premium and ideal
+diamonds. In order to do this, we can first calculate the average carat for
+each color and cut of diamonds, then make a pivot table to show the results:
+
+```{python}
+expr = (
+    diamonds.group_by(["cut", "color"])
+    .agg(carat=_.carat.mean())
+    .pivot_wider(
+        names=("Premium", "Ideal"), names_from="cut", values_from="carat", names_sort=True, values_agg="mean"
+    )
+)
```

Comment:
```suggestion
from ibis import _

expr = (
    diamonds.group_by(["cut", "color"])
    .agg(carat=_.carat.mean())
    .pivot_wider(
        names=("Premium", "Ideal"), names_from="cut", values_from="carat", names_sort=True, values_agg="mean"
    )
)
```
I think this will fail without the import